core/src/impl/Kokkos_Combined_Reducer.hpp

```diff
@@ -569,9 +569,6 @@ auto parallel_reduce(std::string const& label, PolicyType const& policy,
       functor, space_type{}, returnType1, returnType2, returnTypes...);
 
   using combined_functor_type = decltype(combined_functor);
-  static_assert(
-      Impl::FunctorDeclaresValueType<combined_functor_type, void>::value,
-      "value_type not properly detected");
```

Comment:
It's an intern check so I didn't feel that much required to keep it since there is no other place where we really need to ask if a functor declares value type or not outside `FunctorAnalysis`. I'm happy to use the detection idiom just here, though.